### PR TITLE
Fixed a TypeError that triggered when reading a pcap file containing a NTP packet.

### PR DESCRIPTION
**Pull request by Neurocinetics**
Fixed an error (TypeError) that triggered when reading a pcap file that contained ntp packets. The type of variable "cls" was string not a class type (invalid argument for method: issubclass()). By importing the type 'TCP' from 'scapy.layers.inet' and by converting cls to a classtype by reading the key from globals, the TypeError is mitigated.

### CODE REVIEW

**Comment by guedou** [comment]
Can you provide an example of the packet that cause the issue? Also, please add a unit test.

File: scapy/layers/ntp.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 StrFixedLenEnumField, XStrFixedLenField, PacketField, PacketLenField,\
 PacketListField, FieldListField, ConditionalField, PadField)
 from scapy.layers.inet6 import IP6Field
-from scapy.layers.inet import UDP
+from scapy.layers.inet import UDP, TCP, Ether
```
Comment (p-l-) [inline scapy/layers/ntp.py]: That's not needed.

File: scapy/layers/ntp.py [hunk 2]
```diff
@@ -214,7 +214,7 @@ def haslayer(self, cls):
         if cls == "NTP":
             if isinstance(self, NTP):
                 return True
-        elif issubclass(cls, NTP):
+        elif issubclass(globals()[cls], NTP):
```
Comment (p-l-) [inline scapy/layers/ntp.py]: The best way to fix that is
```python
    elif not isinstance(cls, str) and issubclass(cls, NTP):
```

File: scapy/layers/ntp.py (hunk 1 above)
Comment (p-l-) [inline scapy/layers/ntp.py]: That's not needed.

File: scapy/layers/ntp.py (hunk 2 above)
Comment (Neurocinetics) [reply scapy/layers/ntp.py]: Looks much better, thanks.

File: scapy/layers/ntp.py (hunk 1 above)
Comment (Neurocinetics) [reply scapy/layers/ntp.py]: Agreed, my bad.

**Comment by Neurocinetics** [comment]
What is the best place to leave the silly tests I've added?

**Comment by p-l-** [comment]
You can move your tests to `test/regression.uts`. You don't need to load the NTP module, as it is loaded by default. Also, can you squash your first and third commits together?

**Comment by p-l-** [comment]
I don't think we need the `ntp.uts` file any more, do we?

**Comment by Neurocinetics** [comment]
I didn't notice it checking my rebase. Sorry! I Will fix it in a little while.
________________________________

**Comment by guedou** [comment]
Thanks for your PR!

**Comment by karpierz** [comment]
FYI:
I'm just during final testing of "unification" :) #1223 and #1186 and my own previous fix (not published ealier).
Take a look at: #1225 (Im am working on complete and check appropriate tests).

**Pull request merged**